src/features/project/Project.controller.ts

```diff
@@ -0,0 +1,52 @@
+import * as yup from 'yup';
+import { Request, Response } from 'express';
+import { Response200, SendErrorResponse } from '../../utils/Response';
+import { ProjectService } from './Project.service';
+import { projectSchema } from '../../helpers/validations/project';
+
+interface MulterRequest extends Request {
+  file: any;
+}
+
+export class ProjectController {
+  private projectService: ProjectService;
+
+  constructor() {
+    this.projectService = new ProjectService();
+  }
+
+  async addProject(req: MulterRequest, res: Response) {
+    try {
+      //   console.log(req['user']);
+      const {
+        name,
+        description,
+        startDate,
+        endDate,
+        goalAmount,
+        fundRaised,
+        aboutTheCause,
+        planOfAction,
+      } = req.body;
+      await projectSchema.validate(req.body);
+      const file = req.file.filename;
+      const result = await this.projectService.addProject({
+        name,
+        description,
+        startDate,
+        endDate,
+        goalAmount,
+        fundRaised,
+        aboutTheCause,
+        planOfAction,
+        file,
+      });
```

Comment:
You don't have to do this way.

You can use spread operator.

```suggestion
      const result = await this.projectService.addProject({
        ...req.body,
        file,
      });
```